docs/develop/parent-to-dialog.md

```diff
@@ -1,20 +1,20 @@
 ---
 title: Alternative ways of passing messages to a dialog box from its host page
 description: Learn workarounds to use when the messageChild method isn't supported.
-ms.date: 08/31/2023
+ms.date: 11/16/2023
 ms.topic: how-to
 ms.localizationpriority: medium
 ---
 
 # Alternative ways of passing messages to a dialog box from its host page
 
-The recommended way to pass data and messages from a parent page to a child dialog box is with the `messageChild` method as described in [Use the Office dialog API in your Office Add-ins](dialog-api-in-office-add-ins.md#pass-information-to-the-dialog-box). If your add-in is running on a platform or host that doesn't support the [DialogApi 1.2 requirement set](/javascript/api/requirement-sets/common/dialog-api-requirement-sets), there are two other ways that you can pass information to the dialog box.
+The recommended way to pass data and messages from a parent page to a child dialog is with the `messageChild` method as described in [Use the Office dialog API in your Office Add-ins](dialog-api-in-office-add-ins.md#pass-information-to-the-dialog-box). If your add-in is running on a platform or host that doesn't support the [DialogApi 1.2 requirement set](/javascript/api/requirement-sets/common/dialog-api-requirement-sets), there are two other ways that you can pass information to the dialog.
 
-- Add query parameters to the URL that is passed to `displayDialogAsync`.
-- Store the information somewhere that is accessible to both the host window and dialog box. The two windows don't share a common session storage (the [Window.sessionStorage](https://developer.mozilla.org/docs/Web/API/Window/sessionStorage) property), but *if they have the same domain* (including port number, if any), they share a common [Local Storage](https://www.w3schools.com/html/html5_webstorage.asp).\*
+- Store the information somewhere accessible to both the host window and dialog. The two windows don't share a common session storage (the [Window.sessionStorage](https://developer.mozilla.org/docs/Web/API/Window/sessionStorage) property), but *if they have the same domain* (including port number, if any), they share a common [local storage](https://www.w3schools.com/html/html5_webstorage.asp).
```

Comment:
The section about "if they have the same domain" might be confusing.
Please consider adding something about primary domain and 3rd party storage partitioning here. Storage partitioning is affecting outlook on web not desktop - because on OWA the panel's origin (example.com) is outside of the top-level which is outlook.com. On desktop panel's primary domain is example.com, it's not part of any other website so localStorage is fine there.
Also localStorage might still be the only reliable way for those who support Outlook 2019 retail version when it's still IE11 and messageChild is not supported (in some cases like redirect auth flows with panel when we can store something on the backend yet)

Reply:
We add a note after this which mentions storage partitioning and the suggested workarounds (see docs/includes/browser-security-updates.md of this PR).

Reply:
awesome
still please consider that Outlook 2019 honorable mention somewhere - IMHO localStorage is the only option to send message from dialog to panel there

Reply:
Are you referring to the Outlook 2019 retail perpetual license or volume-licensed perpetual Outlook 2019? Based on the [Outlook client support table](https://learn.microsoft.com/javascript/api/requirement-sets/outlook/outlook-api-requirement-sets#outlook-client-support), Outlook 2016 and later retail licenses support up to Mailbox requirement set 1.13. [messageChild](https://learn.microsoft.com/javascript/api/office/office.dialog#office-office-dialog-messagechild-member(1)) was introduced in Mailbox requirement set 1.9.

Reply:
I was referring to the volume licensed Outlook 2019. Addins are running in IE there. messageChild is not working.
Office.context.requirements.isSetSupported('DialogAPI', '1.2') is false there can't use messageChild, local storage works fine though.

Reply:
Please not that on this page here https://learn.microsoft.com/en-us/office/dev/add-ins/develop/dialog-api-in-office-add-ins#conditional-messaging-from-parent-page-to-dialog-box there is a note
"In some situations, the messageChild API, which is a part of the [DialogApi 1.2 requirement set](https://learn.microsoft.com/en-us/javascript/api/requirement-sets/common/dialog-api-requirement-sets), may not be supported. Some alternative ways for parent-to-dialog-box messaging are described in [Alternative ways of passing messages to a dialog box from its host page](https://learn.microsoft.com/en-us/office/dev/add-ins/develop/parent-to-dialog)."
Few months ago I was in this situation, volume licensed Outlook 2019, no messageChild, and now I have a safe localStorage workaround for that scenario.

and sorry, I said in my prvious comment, that localStorage is the only option to send messages from panel to dialog, There are obviously other ways, but localStorage is quite convenient.

Reply:
Thanks again for your feedback, @milczuk. I've updated the note in [Use the Office dialog API in Office Add-ins](https://learn.microsoft.com/office/dev/add-ins/develop/dialog-api-in-office-add-ins#conditional-messaging-from-parent-page-to-dialog-box) to mention that the `messageChild` API isn't supported in volume-licensed perpetual Outlook 2016 and 2019. It then links to [Alternative ways of passing messages to a dialog box from its host page](https://learn.microsoft.com/office/dev/add-ins/develop/parent-to-dialog) where the local storage option is mentioned.